src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -56,16 +54,34 @@ public bool TryRegister(SafeSocketHandle socket, out Interop.Error error)
 
         private static readonly object s_lock = new object();
 
-        // In debug builds, force there to be 2 engines. In release builds, use half the number of processors when
-        // there are at least 6. The lower bound is to avoid using multiple engines on systems which aren't servers.
-#pragma warning disable CA1802 // const works for debug, but needs to be static readonly for release
-        private static readonly int s_engineCount =
-#if DEBUG
-            2;
-#else
-            Environment.ProcessorCount >= 6 ? Environment.ProcessorCount / 2 : 1;
-#endif
-#pragma warning restore CA1802
+        private static readonly int s_engineCount = GetEnginesCount();
```

Comment:
Nit: should this be s_maxEngineCount?  We won't always have this many, but we may grow to this many based on the number of concurrent sockets, right?

Reply:
@stephentoub I've suggested to remove that logic as part of this PR (https://github.com/dotnet/runtime/pull/35800#issuecomment-624861323).

Reply:
@tmds You are most probably right. The only use case for keeping it is a machine with many cores and very few connections. Which should be uncommon.

Would you prefer me to remove it now or would you like to do this in your upcoming PR that is going to enable the "inlining"?

Reply:
@adamsitnik remove it here, it is unrelated to inlining.

Reply:
@tmds I am going to merge it as it is right now as I would really love to see the update numbers. I am going to send a PR with MinHandles logic removal today or tomorrow.